library/L2_HAL/communication/test/esp8266_test.cpp

```diff
@@ -19,7 +19,46 @@ TEST_CASE("Esp8266 Test", "[esp8266]")
 
   SECTION("Test Module")
   {
+    auto uart_response =
+        [](uint8_t * data,
+           size_t size,
+           std::chrono::microseconds timeout) -> Status {
+      static constexpr uint8_t kConfirmationResponse[] = "\r\nOK";
+      static uint8_t count = 0;
+
+      // After writing the test command to the wifi module, the microcontroller
+      // is supposed to read a confirmation response
+      if (count == 0)
+      {
+        CHECK(timeout == Esp8266::kReadTimeout);
+        CHECK(size == (sizeof(kConfirmationResponse) - 1));
+
+        for (size_t i = 0; i < size; i++)
+        {
+          data[i] = kConfirmationResponse[i];
+        }
+        count++;
+      }
+      // The second uart read is used to flush the serial buffer
+      else
+      {
+        CHECK(timeout == Esp8266::kReadTimeout);
+        CHECK(size == 1);
+      }
```

Comment:
Can you add an issue to convert this into a template test, or something more generic. You echo this multiple times throughout your test code.

Reply:
Alright